src/vg_active_segment.erl

```diff
@@ -29,18 +30,33 @@
                 index_fd    :: file:fd(),
                 topic       :: binary(),
                 partition   :: integer(),
-
+                history_tab :: atom(),
                 config      :: #config{}
                }).
 
--define(SERVER(Topic, Partition), {via, gproc, {n,l,{Topic,Partition}}}). % binary_to_atom(<<Topic/binary, Partition>>, utf8)).
+-define(SERVER(Topic, Partition), {via, gproc, {n,l,{Topic,Partition}}}).
+-define(TAB(Topic, Partition), binary_to_atom(<<Topic/binary, Partition>>, utf8)).
+-define(TAB_OPTIONS, [public, named_table, bag, {write_concurrency, true}]).
 
 start_link(Topic, Partition, NextBrick) ->
+    Tab = ?TAB(Topic, Partition),
+    case ets:info(Tab, name) of
+        undefined ->
+            ets:new(Tab, ?TAB_OPTIONS);
+        _ ->
+            ok
+    end,
+
     gen_server:start_link(?SERVER(Topic, Partition), ?MODULE, [Topic, Partition, NextBrick], []).
 
 write(Topic, Partition, MessageSet) ->
     gen_server:call(?SERVER(Topic, Partition), {write, MessageSet}).
 
+ack(Topic, Partition, LatestId) ->
+    ets:select_delete(?TAB(Topic, Partition), [{{'$1', '$2'},
```

Comment:
http://erlang.org/doc/man/ets.html#delete-2

I feel like I've spent a lot of time wrestling with select_* related performance problems.  Not a big deal right now.